draft-ietf-quic-transport.md

```diff
@@ -3838,143 +3838,135 @@ The details of loss detection and congestion control are described in
+first.
+
+An endpoint thus attempts to use ECN and validates this for each new connection,
+when switching to a server's preferred address, and on active connection
+migration to a new path.  {{ecn-alg}} describes one possible algorithm.
 
 Other methods of probing paths for ECN support are possible, as are different
 marking strategies. Implementations MAY use other methods defined in RFCs; see
-{{?RFC8311}}. Implementations that use the ECT(1) codepoint need to perform ECN
-validation using ECT(1) counts.
+{{?RFC8311}}. Implementations that use the ECT(1) codepoint need to
+perform ECN validation using the reported ECT(1) counts.
 
 
-#### Receiving ACK Frames {#ecn-ack}
+#### Receiving ACK Frames with ECN Counts {#ecn-ack}
 
-Erroneous application of ECN marks in the network can result in degraded
-connection performance.  An endpoint that receives an ACK frame with ECN
-counts therefore validates the counts before using them. It performs this
-validation by comparing newly received counts against those from the last
-successfully processed ACK frame. Any increase in ECN counts is validated
-based on the markings that were applied to packets that are newly
-acknowledged in the ACK frame.
+Erroneous application of CE-marks by the network can result in degraded
+connection performance.  An endpoint that receives an ACK frame with ECN counts
+therefore validates the counts before using them. It performs this validation by
+comparing newly received counts against those from the last successfully
+processed ACK frame. Any increase in the ECN counts is validated based on the
+ECN markings that were applied to packets that are newly acknowledged in the ACK
+frame.
 
-If an ACK frame newly acknowledges a packet that the endpoint sent with either
-ECT(0) or ECT(1) codepoints set, ECN validation fails if ECN counts are not
-present in the ACK frame.  This check detects a network element that zeroes out
-ECN bits or a peer that is unable to access ECN markings.
+If an ACK frame newly acknowledges a packet that the endpoint sent setting
+either the ECT(0) or ECT(1) codepoint, ECN validation fails if the ECN counts
+are not present in the ACK frame. This check detects a network element that
+zeroes the ECN field or a peer that is unable to access ECN the markings.
 
-ECN validation fails if the sum of the increase in ECT(0) and ECN-CE counts is
-less than the number of newly acknowledged packets that were originally sent
+ECN validation also fails if the sum of the increase in ECT(0) and ECN-CE counts
+is less than the number of newly acknowledged packets that were originally sent
 with an ECT(0) marking.  Similarly, ECN validation fails if the sum of the
 increases to ECT(1) and ECN-CE counts is less than the number of newly
 acknowledged packets sent with an ECT(1) marking.  These checks can detect
-removal of ECN markings in the network.
+remarking of ECN CE-markings by the network.
```

Comment:
I know that "remarking" is likely a term of art, but it is also a word with a completely different meaning.  I think we should avoid it.

Reply:
"These checks can detect network elements that overwrite ECN-CE markings with an ECT(0) or ECT(1) codepoint."

Reply:
I'm OK with a change of word, but the checks only detect some changes to the ECN-Field. They can't detect rewriting of a CE-Marking back to ECT(0) (for instance when ECT(0) was sent), so we should be careful to avoid giving that impression..